functests/dpdk/dpdk.go

```diff
@@ -997,14 +1019,51 @@ func waitForSRIOVStable() {
 	}, waitingTime, 1*time.Second).Should(BeTrue())
 }
 
-func filterPerformanceProfilesWithAvailableNodes(performanceProfiles []*perfv1alpha1.PerformanceProfile, nodeSelector map[string]string) []*perfv1alpha1.PerformanceProfile {
-	availableProfiles := make([]*perfv1alpha1.PerformanceProfile, 0)
+// findPerformanceProfileAndPolicyWithAvailableNodes finds a profile/sriovPolicy match for which a node with
+// allocatable resources is available. It will return a profile/sriovPolicy for a policy with resource name
+// "dpdknic", or a pair with the most available resource on node
+func findPerformanceProfileAndPolicyWithAvailableNodes(performanceProfiles []*perfv1alpha1.PerformanceProfile, nodeSelector map[string]string,
+) (preferredProfile *perfv1alpha1.PerformanceProfile, preferredResource string, preferredDevice *sriovv1.InterfaceExt, err error) {
+	currentResourceCount := 0
+	sriovInfos, err := sriovcluster.DiscoverSriov(sriovclient, SRIOV_OPERATOR_NAMESPACE)
+	Expect(err).ToNot(HaveOccurred())
+
+	sriovPolicies := &sriovv1.SriovNetworkNodePolicyList{}
+	err = client.Client.List(context.TODO(), sriovPolicies, &goclient.ListOptions{Namespace: SRIOV_OPERATOR_NAMESPACE})
+	Expect(err).ToNot(HaveOccurred())
+
 	for _, profile := range performanceProfiles {
 		profileNodeSelector := nodes.NodesSelectorUnion(nodeSelector, profile.Spec.NodeSelector)
```

Comment:
Didn't notice before, but `NodesSelectorUnion` should be intersection?
I would also remove the initial `Nodes`, see my other comment below.

Reply:
The selector should be an union. We want to have the nodes that are limited by both the provided selector, and the selector of the profile.
The nodes we get as a result will be an intersection.

Reply:
Got it, thanks.
Can you rename the func to `SelectiorUnion` as the nodes is already implicit in the package name?